CONTRIBUTING.md

```diff
@@ -3,10 +3,79 @@ There is a white-list for contributors who have signed the OCA so please add a c
 from your github profile.
 
-Pull requests can only be merged by committers and a committer cannot merge
-his or her own pull requests.
+It is desirable to review all changes by at least one other person. When creating a pull request,
+make sure you designate an assignee to ensure the request is taken care of in a timely manner.
+The assignee will guide you trough the process and find the most suitable review style for your
+change.
 
-When creating a pull request, make sure you designate an assignee to ensure
-the request is processed in a timely manner.
+Reviews
+=======
+
+The **Truffle** repository is primarily about APIs and not *only* the code. It is a framework that
+other (often unknown) developers consume and rely on. As such we need to adhere to rules of 
+[API design][3] when making changes. We categorize the changes into three groups:
+- trivial API change
+- complex API change
+- no API change
+
+Each category requires different level of attention specified below. The process and its terminology
+models the [review process][2] used by NetBeans. Its most important points are highlighted in this 
+document.
+
+An API isn't just what is written in Javadoc, it is not just about the [method signatures][4]. 
+The API includes properties, files, ports, protocols that are being read or written to.
+It includes L10N, packages, versioning. It includes runtime behavior, threading, 
+memory management, etc. In short API is *everything somebody else can depend on*.
+
+Trivial API Change
+------------------
+
+It is very likely that by changing something in the Truffle repository, you are changing an API. If you
+do so, it is important to stick with essential properties of a good API change:
+
+- be backward compatible
+- be well documented
+- have sufficient test coverage
+- be ready for future evolution
+
+If your change satisfies these properties you are eligible for a **fast-track** review: create a pull request,
+let people know about it, assign a responsible reviewer, react to *requests for change* and after a week 
+you should be ready to merge. Announce the *final 24 hour count down* and then merge or convince an
+approved committer to merge.
+
+Trivial change is usually about single method/class addition or other *little improvements* of existing APIs. 
+Additions of whole packages very likely fall into complex change category. In case of disagreement any reviewer
+can turn a trivial change into complex one by requesting **standard** review.
+
+Complex API Change
+------------------
+
+If an API change is inherently backward *incompatible*, or if it is huge, it is subject to **standard** review.
+The purpose of such review is to seek wider consensus among involved parties. The ideal state of such review
+should:
+
+- describe the intention, use-cases and let everyone comment
+- nominate four "voting" reviewers
+- have an *inception call* where reviewers approve/reject the project and set constraints/directions
```

Comment:
but then by your description it has to happen before the pull request, so it's irrelevant imho